server/upsd.c

```diff
@@ -303,11 +466,17 @@ static void setuptcp(stype_t *server)
 			fatal_with_errno(EXIT_FAILURE, "setuptcp: setsockopt");
 		}
 
-		/* Ordinarily we request that IPv6 listeners handle only IPv6.
+		/* Ordinarily we request that IPv6 listeners handle only IPv6
+		 * (except when we handle `LISTEN *` as detailed above).
+		 * Note we specifically try to ensure this when CLI requires
+		 * IPv6-only behavior (even if we want "any" addr for `LISTEN *`).
 		 * TOTHINK: Does any platform need `#ifdef IPV6_V6ONLY` given
 		 * that we apparently already have AF_INET6 OS support everywhere?
+		 * TOTHINK: Do we want to setsockopt() to explicitly allow dual-stack
+		 * (perhaps counteracting OS default or customized configuration)
+		 * when handling `LISTEN *` use-cases?
```

Comment:
No,we absolutely never want to turn on mapped addresses if the system default has them turned off.   The modern BSD world views use of mapped addresses as a bug.  It was a transition mechanism to accomodate server code that had one socket, and had not yet been updated to be multi-AF aware.